jgrapht-core/src/main/java/org/jgrapht/alg/isomorphism/AHUUnrootedTreeIsomorphismInspector.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * (C) Copyright 2018-2018, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.isomorphism;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphMapping;
+import org.jgrapht.alg.shortestpath.TreeMeasurer;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Objects;
+
+/**
+ * This is an implementation of the AHU algorithm for detecting an (unweighted) isomorphism between two unrooted trees.
+ * Please see <a href="http://mathworld.wolfram.com/GraphIsomorphism.html">mathworld.wolfram.com</a> for a complete
+ * definition of the isomorphism problem for general graphs.
+ *
```

Comment:
If the input is directed, then either that should be a prerequisite violation, or you should find the root (which is easy to do since there should be exactly one vertex with indegree 0).